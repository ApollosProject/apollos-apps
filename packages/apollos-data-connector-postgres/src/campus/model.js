import { DataTypes } from 'sequelize';
import { defineModel, configureModel } from '../postgres';

const createModel = defineModel({
  modelName: 'campus',
  resolveType: () => 'Campus',
  external: true,
  attributes: {
    name: DataTypes.TEXT,
    street1: DataTypes.TEXT,
    street2: DataTypes.TEXT,
    city: DataTypes.TEXT,
    state: DataTypes.TEXT,
    postalCode: DataTypes.TEXT,
    latitude: DataTypes.FLOAT,
    longitude: DataTypes.FLOAT,
    imageUrl: DataTypes.TEXT,
    digital: DataTypes.BOOLEAN,
  },
});

const setupModel = configureModel(({ sequelize }) => {
  sequelize.models.campus.hasMany(sequelize.models.people);
  sequelize.models.people.belongsTo(sequelize.models.campus);
});

export { createModel, setupModel };
