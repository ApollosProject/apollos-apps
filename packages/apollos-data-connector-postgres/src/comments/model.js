import { DataTypes } from 'sequelize';
import { defineModel, configureModel } from '../postgres';

const Visibility = {
  PUBLIC: 'PUBLIC',
  PRIVATE: 'PRIVATE',
  FOLLOWERS: 'FOLLOWERS',
};

const createModel = defineModel({
  modelName: 'comments',
  resolveType: () => 'Comment',
  attributes: {
    text: DataTypes.TEXT,
    visibility: DataTypes.ENUM(Object.values(Visibility)),
    externalParentId: DataTypes.TEXT,
    externalParentType: DataTypes.TEXT,
    externalParentSource: DataTypes.TEXT,
    flagCount: {
      type: DataTypes.INTEGER,
      defaultValue: 0,
    },
  },
});

const setupModel = configureModel(({ sequelize }) => {
  sequelize.models.comments.belongsTo(sequelize.models.people);
  sequelize.models.people.hasMany(sequelize.models.comments);

  // Used to identify user relationships when querying for comments
  // Specifically, to sort the comments of those you follow to the top
  sequelize.models.comments.hasMany(sequelize.models.follows, {
    foreignKey: 'followedPersonId',
    sourceKey: 'personId',
    constraints: false,
  });
});

export { createModel, Visibility, setupModel };
