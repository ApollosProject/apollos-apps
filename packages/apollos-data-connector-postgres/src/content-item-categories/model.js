import { DataTypes } from 'sequelize';
import { defineModel, configureModel } from '../postgres';

const createModel = defineModel({
  modelName: 'contentItemCategory',
  resolveType: () => 'ContentItemCategory',
  external: true,
  attributes: {
    title: DataTypes.TEXT,
  },
});

const setupModel = configureModel(({ sequelize }) => {
  sequelize.models.contentItem.belongsTo(sequelize.models.contentItemCategory);
  sequelize.models.contentItemCategory.hasMany(sequelize.models.contentItem);
});

export { createModel, setupModel };
