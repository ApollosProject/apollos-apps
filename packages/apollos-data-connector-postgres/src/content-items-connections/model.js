import { DataTypes } from 'sequelize';
import { defineModel, configureModel } from '../postgres';

const createModel = defineModel({
  modelName: 'contentItemsConnection',
  resolveType: () => 'ContentItemConnection', // shouldn't get called, ContentItem will be set on the shovel.
  external: true,
  attributes: {
    order: DataTypes.INTEGER,
  },
});

const setupModel = configureModel(({ sequelize }) => {
  sequelize.models.contentItem.belongsToMany(sequelize.models.contentItem, {
    through: sequelize.models.contentItemsConnection,
    as: 'children',
    foreignKey: 'parentId',
    otherKey: 'childId',
  });

  sequelize.models.contentItem.belongsToMany(sequelize.models.contentItem, {
    through: sequelize.models.contentItemsConnection,
    as: 'parents',
    otherKey: 'parentId',
    foreignKey: 'childId',
  });
});

export { createModel, setupModel };
