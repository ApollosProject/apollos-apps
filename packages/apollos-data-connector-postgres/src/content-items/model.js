import { DataTypes } from 'sequelize';
import { defineModel, configureModel } from '../postgres';

const createModel = defineModel({
  modelName: 'contentItem',
  resolveType: () => 'UniversalContentItem', // shouldn't get called, ContentItem will be set on the shovel.
  external: true,
  attributes: {
    title: DataTypes.TEXT,
    summary: DataTypes.TEXT,
    htmlContent: DataTypes.TEXT,
    publishAt: DataTypes.DATE,
    active: DataTypes.BOOLEAN,
  },
});

const setupModel = configureModel(({ sequelize }) => {
  sequelize.models.contentItem.belongsTo(sequelize.models.media, {
    as: 'coverImage',
    foreignKey: 'coverImageId',
  });

  sequelize.models.contentItem.belongsTo(sequelize.models.contentItem, {
    as: 'parent',
    foreignKey: 'parentId',
  });
});

export { createModel, setupModel };
