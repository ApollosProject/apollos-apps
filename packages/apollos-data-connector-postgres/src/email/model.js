const createModel = () => {};

const setupModel = () => {};

export { createModel, setupModel };
