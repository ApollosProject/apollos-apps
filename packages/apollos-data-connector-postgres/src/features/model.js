import { DataTypes } from 'sequelize';
import { defineModel, configureModel } from '../postgres';

const createModel = defineModel({
  modelName: 'feature',
  resolveType: () => 'Feature',
  attributes: {
    parentId: DataTypes.UUID,
    parentType: DataTypes.TEXT,
    data: DataTypes.JSONB,
    type: DataTypes.TEXT,
  },
  sequelizeOptions: {
    tableName: 'feature',
    underscored: true,
  },
});

const setupModel = configureModel(({ sequelize }) => {
  sequelize.models.feature.belongsTo(sequelize.models.contentItem, {
    foreignKey: 'parentId',
  });
  sequelize.models.contentItem.hasMany(sequelize.models.feature, {
    foreignKey: 'parentId',
    scope: { parentType: 'ContentItem' },
  });
});

export { createModel, setupModel };
