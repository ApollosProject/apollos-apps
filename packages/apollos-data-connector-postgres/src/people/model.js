import { DataTypes } from 'sequelize';
import { defineModel } from '../postgres';

const Gender = {
  MALE: 'MALE',
  FEMALE: 'FEMALE',
  UNKNOWN: 'UNKNOWN',
};

const createModel = defineModel({
  modelName: 'people',
  resolveType: () => 'Person',
  external: true,
  attributes: {
    firstName: DataTypes.STRING,
    lastName: DataTypes.STRING,
    email: DataTypes.STRING,
    birthDate: DataTypes.DATE,
    profileImageUrl: DataTypes.TEXT,
    gender: DataTypes.ENUM(Object.values(Gender)),
    apollosUser: {
      type: DataTypes.BOOLEAN,
      defaultValue: false,
    },
  },
});

export { createModel };
