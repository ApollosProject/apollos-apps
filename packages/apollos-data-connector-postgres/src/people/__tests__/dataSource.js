import { sequelize, sync } from '../../postgres/index';
import { createModel } from '../model';
import PeopleDataSource from '../dataSource';

let personId;

const context = {
  dataSources: {
    Auth: {
      getCurrentPerson: () => ({ id: personId }),
      ORIGIN_TYPE: 'rock',
    },
    BinaryFiles: {
      uploadFile: jest.fn(() => Promise.resolve('456')),
      findOrReturnImageUrl: () => 'https://the.image.com/bop.jaz',
    },
  },
};

describe('Apollos Postgres People DataSource', () => {
  let peopleDataSource;

  beforeEach(async () => {
    personId = 1;

    await createModel();
    await sync();

    peopleDataSource = new PeopleDataSource();
    peopleDataSource.initialize({ context });
  });

  afterEach(async () => {
    await sequelize.drop({});
  });

  it('should find a user by a rock id', async () => {
    await peopleDataSource.model.create({
      originId: '1',
      originType: 'rock',
      firstName: 'Vincent',
      lastName: 'Wilson',
    });

    const person = await peopleDataSource.getFromId(1, null, {
      originType: 'rock',
    });

    expect(person.firstName).toBe('Vincent');
    expect(person.originId).toBe('1');
    expect(person.id).toBeDefined();
  });

  it('should find a user by postgres id', async () => {
    const newPerson = await peopleDataSource.model.create({
      originId: '1',
      originType: 'rock',
      firstName: 'John',
      lastName: 'Williams',
    });

    const person = await peopleDataSource.getFromId(newPerson.id);

    expect(person.firstName).toBe('John');
    expect(person.originId).toBe('1');
    expect(person.id).toBe(newPerson.id);
  });

  it('should create a user', async () => {
    const newPersonId = await peopleDataSource.create({
      FirstName: 'Vincent',
      lastName: 'Vincent',
      Gender: 'Male',
      originId: 1,
      originType: 'rock',
    });

    const newPerson = await peopleDataSource.getFromId(newPersonId);

    expect(newPerson.firstName).toBe('Vincent');
    expect(newPerson.gender).toBe('MALE');
  });

  it('update a users attributes', async () => {
    await peopleDataSource.model.create({
      originId: '1',
      originType: 'rock',
      firstName: 'John',
      lastName: 'Williams',
    });

    personId = 1;
    const person = await peopleDataSource.updateProfile([
      { field: 'FirstName', value: 'Milton' },
    ]);

    expect(person.firstName).toBe('Milton');

    const updatedPerson = await peopleDataSource.getFromId(person.id);

    expect(updatedPerson.firstName).toBe('Milton');
  });

  it("uploads a user's profile picture", async () => {
    await peopleDataSource.model.create({
      originId: '1',
      originType: 'rock',
      firstName: 'John',
      lastName: 'Williams',
    });

    personId = 1;
    const person = await peopleDataSource.uploadProfileImage(
      { createReadStream: () => '123', filename: 'img.jpg' },
      456
    );

    expect(person.profileImageUrl).toBe('https://the.image.com/bop.jaz');

    const updatedPerson = await peopleDataSource.getFromId(person.id);

    expect(updatedPerson.profileImageUrl).toBe('https://the.image.com/bop.jaz');
  });
});
