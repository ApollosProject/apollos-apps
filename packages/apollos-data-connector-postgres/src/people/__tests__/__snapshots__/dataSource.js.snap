// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apollos Postgres People DataSource should throw an error if current user isnt in the db 1`] = `"Invalid Credentials"`;
